app/Http/Controllers/DemandController.php

```diff
@@ -196,34 +196,8 @@ public function click(Request $request, $id)
         //     ]);
         // }
 
-        $response = new \Symfony\Component\HttpFoundation\Response($url);
+      return  new RedirectResponse($url);
 
-        // last click id will be used to track conversions
-        $response->headers->setCookie(
-            new \Symfony\Component\HttpFoundation\Cookie(
-                'cid',
-                $request->query->get('cid'),
-                new \DateTime('+ 1 month')
-            )
-        );
-
-        $response->setContent(
-            sprintf(
-                '<!DOCTYPE html>
-<html>
-  <head>
-      <meta charset="UTF-8" />
-      <meta http-equiv="refresh" content="5;url=%1$s" />
-
-      <title>Redirecting to %1$s</title>
-  </head>
-  <body>
-      Redirecting to <a href="%1$s">%1$s</a>.
-  </body>
-</html>',
-                htmlspecialchars($url, ENT_QUOTES, 'UTF-8')
-            )
-        );
 
         return $response;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Review the data-flow - use of uninitialized value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS836)